src/sentry/search/events/filter.py

```diff
@@ -1129,3 +1131,25 @@ def _issue_filter_converter(self, search_filter: SearchFilter) -> Optional[Where
                 SearchValue(filter_values if search_filter.is_in_filter else filter_values[0]),
             )
         )
+
+    def _transaction_status_filter_converter(
+        self, search_filter: SearchFilter
+    ) -> Optional[WhereType]:
+        # Handle "has" queries
+        if search_filter.value.raw_value == "":
+            return Condition(
+                Function("isNull", [self.resolve_field_alias(search_filter.key.name)]),
+                Op(search_filter.operator),
+                1,
```

Comment:
Depending on the op we could construct this a bit cleaner as
```python
Condition(self.resolve_field_alias(search_filter.key.name), Op.IS_NOT_NULL)
```

Reply:
Hey Will, I noticed that this doesn't work when using negations, ie: `!has:transaction.status` is treated the same as `has:transaction.status` during my testing. The original seems to work correctly for negative and positive cases. I'll try to dig into why this is the case but I think I may keep this part unchanged. What do you think?

Reply:
ah you have to change the `Op.IS_NOT_NULL` to `Op.IS_NULL` depending on the `search_filter.operator` probably, can discuss further in slack if you need further clarification

Reply:
Ahh you're right, I don't know why but I didn't think `search_filter.operator` was reflecting the negative. Updated!